br/pkg/lightning/common/common.go

```diff
@@ -89,17 +89,26 @@ func getGlobalAutoIDAlloc(store kv.Storage, dbID int64, tblInfo *model.TableInfo
 	hasAutoIncID := tblInfo.GetAutoIncrementColInfo() != nil
 	hasAutoRandID := tblInfo.ContainsAutoRandomBits()
 
-	// Current TiDB has some limitations for auto ID.
+	// Before MySQL compatible AUTO_INCREMENT, TiDB has some limitations:
 	// 1. Auto increment ID and auto row ID are using the same RowID allocator.
 	//    See https://github.com/pingcap/tidb/issues/982.
 	// 2. Auto random column must be a clustered primary key. That is to say,
 	//    there is no implicit row ID for tables with auto random column.
 	// 3. There is at most one auto column in a table.
 	// Therefore, we assume there is only one auto column in a table and use RowID allocator if possible.
+	//
+	// If AUTO_ID_CACHE=1 is set (i.e. MySQL compatible AUTO_INCREMENT), the limitations do not hold anymore.
 	switch {
-	case hasRowID || hasAutoIncID:
+	case hasRowID:
 		return autoid.NewAllocator(store, dbID, tblInfo.ID, tblInfo.IsAutoIncColUnsigned(),
 			autoid.RowIDAllocType, noCache, tblVer), nil
+	case hasAutoIncID:
+		allocType := autoid.RowIDAllocType
+		if tblInfo.SepAutoInc() {
+			allocType = autoid.AutoIncrementType
```

Comment:
seems for lightning only the type is changed, and the allocated ID of import data is also changed. What's the new behaviour of this new type? and I'd ask if the data has non-continuous AUTO INCREMENT ID after import is it acceptable?

Reply:
> What's the new behaviour of this new type?

For lightning, this change is not related to the fix of #46093 
I make this change just because this is the correct behaviour, when `SepAutoInc() == true`, the allocator should use the autoid's meta key rather than rowid's meta key.


> and I'd ask if the data has non-continuous AUTO INCREMENT ID after import is it acceptable?

This is because the cached ids in the memory of auto id service is dropped, BR get the current id by reading the auto id's meta key.
If the auto id service panic unexpecitally, we also lose some cached ids, and there would be holes in the id sequence. I think that's acceptable.